packages/ckeditor5-link/src/linkcommand.ts

```diff
@@ -277,4 +288,37 @@ export default class LinkCommand extends Command {
 
 		return true;
 	}
+
+	private _updateLinkContent( model: Model, writer: Writer, range: Range, href: string ): Range {
+		const attributes = toMap( {} );
+
+		attributes.set( 'linkHref', href );
+
+		const text = writer.createText( href, attributes );
+
+		return model.insertContent( text, range );
+	}
+}
+
+function getTextFromRange( range: Range | null ): string {
+	if ( !range || range && !Array.from( range.getItems() ).length ) {
```

Comment:
You could simplify this one

```
const firstNode = first( range.getItems() );

if( !range || range && !firstNode ) { 
	... 
}
```

Reply:
but how we could get items from range until we didn't check is range null or not?
![image](https://user-images.githubusercontent.com/47000454/226287706-184f76df-8c59-4a1f-a3dd-1701f8745185.png)

Reply:
TS is meant to help us but it's not all-knowing. We know that the model document selection in the editor has always at least one range and one position. There's no scenario when document selection has no ranges. We can use this to `!` certain things and simplify code.

Reply:
This code could be simplified to

```js
function getTextFromRange( range: Range ): string | null {
	if ( range.isCollapsed ) {
        return null;
    }

	const firstNode = first( range.getItems() );

	if ( firstNode.is( '$text' ) || firstNode.is( '$textProxy' ) ) {
		return firstNode.data;
	}

	return null;
}
```

because:
1. you can safely assume there always will be a range, 
2. you don't have to engage `TreeWalker` to know there are no items in the range. A collapsed range has no items.